### PR TITLE
Add one ACK carve-out

### PR DESCRIPTION
**Pull request by tcharding**
In an effort to reduce review burden add so strict rules defining 3 times where we think it is acceptable for code to be merged with a single ACK. Feels a bit like we only just added the refactor carve-out and now we are adding more carve-outs, so whats next? We should take these rules seriously, if we are to be taken seriously.

```
We reserve the right to merge PRs with a single ACK [0], at any time, if they match
any of the following conditions:

1. PR only touches CI i.e, only changes any of the `test.sh` scripts and/or
   stuff in `.github/workflows`.
2. Non-content changing documentation fixes i.e., grammar/typos, spelling, full
   stops, capital letters. Any change with more substance must still get two
   ACKs.
3. Code moves that do not change the API i.e., moving error types to a submodule
   and re-exporting them from the original module.  Must not include any code
   changes except to import paths.

[0] - Obviously author and ACK'er must not be the same person.
```

### CODE REVIEW

**Comment by tcharding** [comment]
If this merges I feel like @apoelstra and I are at most risk of abusing this, we should be careful to not fudge the rules to rush things. Any maintainers uncomfortable with this carve out please say, this should not be seen to be pushed through by me and/or Andrew.

This PR requires buy in from the other maintainers before merge.

**Comment by apoelstra** [comment]
Looks good to me. I think you should clarify the "does not change the API" one to change "to a submodule" to "to a private submodule" or "to a non-exposed submodule" so it's clear that we can't even *add* API surface, even obscure surface, without two acks.

**Comment by tcharding** [comment]
> so it's clear that we can't even add API surface, even obscure surface, without two acks.

Just to clarify, we can add to the API with one ack because of the refactor carve-out (but it requires the two week timeout). I extended rule 3 to be:
```
3. Code moves that do not change the API e.g., moving error types to a private
   submodule and re-exporting them from the original module. Must not include
   any code changes except to import paths. This rule is more restrictive than
   the refactor carve out, requires absolutely no change to the public API.
```

**Comment by apoelstra** [comment]
I'd split the last sentence. As written it is missing a conjunction which could be `which` or `and` ... and the choice of which causes the sentence to have opposite meanings.

```
This rule is more restrictive than the refactor carve-out. It requires absolutely no change to the public API.
```

(Also we should hyphenate carve-out :). See #2045.)

**Comment by tcharding** [comment]
English is definitely harder to write than code :)

**Comment by tcharding** [comment]
Used "carve-out" and split sentence as suggested. Thanks

**Review by sanket1729 — APPROVED** [review]
ACK 4222e4d817d90c3382d3a0ef49a4305100d1857e

**Comment by apoelstra** [comment]
Since this is a policy change I'll leave it open for a week or two (esp as people are at Tabconf) before merging.

Though I will continue to immediately one-ack-merge CI fixes in the meantime.

**Review by RCasatta — APPROVED** [review]
ACK 4222e4d817d90c3382d3a0ef49a4305100d1857e

**Comment by sanket1729** [comment]
I believe this can be merged now.

**Pull request merged**
